apps/extension/src/entry/offscreen-handler.ts

```diff
@@ -28,24 +41,28 @@ chrome.runtime.onMessage.addListener((req, _sender, respond) => {
 });
 
 const spawnActionBuildWorker = (req: ActionBuildRequest) => {
```

Comment:
The changes to this function don't change any functionality, right? They're just a refactor?

Reply:
correct, it's just a switch to the `Promise.withResolvers` syntax, could be reverted.

Reply:
Ah no need to revert, I just wanted to make sure I was understanding it correctly